### DISCUSSION TITLE
Netflow view port being used in traffic

### DISCUSSION THREAD

**Discussion by awisetech** [grafolean/grafolean#44] [General]
Hi,

I just installed this. It was extremely easy to setup and get working. I wanted to know if there was a way to see the port being used in the netflow protocols or somewhere. 

Thanks

**Comment by grafolean** [comment]
Hi, glad to hear that!

Sorry, not sure which port you mean? The port to which your device (e.g. router) is sending netflow data to, the port where Grafolean (actually Grafolean Netflow Bot collector) is expecting this data, or the port as part of the traffic (e.g., showing that your browser is talking to google.com on port 443)?

If the last, then yes - you just need to setup netflow in Grafolean and this data will be visible as part of the flows. (more advanced: it is also collected in Postgres and can be queried directly if that's what you prefer)

**Comment by awisetech** [comment]
Thank you for the reply, I appreciate it. 

Yes, I meant the last. I'd like to see what ports the traffic are using. I do not see it on any of the info from the default netflow template. I guess I'll need to dig deeper. Any tips would be appreciated. :)

**Comment by awisetech** [comment]
Hi,

How can I show what TCP/UDP ports are being used? I just see the IPs.

Thanks

**Comment by grafolean** [comment]
Good question, actually. I checked and only IP and protocol was being sent, but not the port. :-/

Unfortunately, I don't use Grafolean anywhere anymore myself, however, the fix didn't look that difficult, so I just went ahead and implemented it (netflow bot v0.3.6-rc.2). To my surprise the whole build pipeline still worked without problems... nice.

Anyway, to use it, you need to do 3 things:

1) in `docker-compose.yml`, where there is an image for service `netflowbot`, add the version so that it looks like this:
```
  netflowbot:
    image: grafolean/grafolean-netflow-bot:v0.3.6-rc.2
```

2) restart this service (run `docker-compose restart netflowbot`)

3) in UI, open your Netflow dashboard and add a new widget:

![image](https://github.com/grafolean/grafolean/assets/52440604/c4d2a933-e66a-4f76-89a3-2ecf54471fa9)

Value for field `Path filter` must be `netflow.$netflowSelectedInterval.$netflowSelectedDirection.entity.$selectedEntityId.topdstports.?`, for other values feel free to copy them from the image. You can set the number of top candidates as you wish, of course, or change other parameters as you see fit.

Hope it helps, and I hope it is what you had in mind. :)

**Reply by awisetech** [reply]
Wow, I missed this email notification. I will try this. Thank you so much for the reply.
